core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -65,6 +71,19 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
       Future.successful(snapshotRef)
     }
 
+  def validateProtectedStatus(workspaceContext: Workspace, snapshot: NamedDataRepoSnapshot): Unit =
+    // logically it might make more sense to check if the snapshot is protected before the workspace
+    // but that is a more expensive check
+    // check if workspace is protected
+    if (!workspaceContext.bucketName.startsWith("fc-secure")) {
```

Comment:
Same

Reply:
Possibly include a comment explaining that this is a bit of a hack and that this responsibility should be moved to Terra policy service when it's ready?  Bonus points for including a ticket if that work is tracked somewhere.

Reply:
Can we also do the TDR check here for Azure protected workspaces? Or does that happen elsewhere (@aherbst-broad)?

Reply:
@cahrens not sure I follow -- WSM is the service that enforces policy for Azure protected workspaces. I am wondering if there is a way to perform this check in WSM today since these are fundamentally "rawls stage" WSM workspaces that are holding the snapshots. Fine for now though.

Reply:
Perhaps the workspace created does not have a cloud context and therefore no cloud platform to check and so the check passes?

Reply:
Snapshot-by-reference into Azure workspaces doesn't work at all right now - it doesn't fail gracefully but it does fail, which is at least compliant and something we intend to implement in the future.  Snapshot by reference is pretty rare and not a big problem, but we deemed it still worth covering the compliance gap.  More information in [this doc](https://docs.google.com/document/d/1lxmWErJczgdCZabGRNFQDD2We6afe4YhdV4HF2bHLGQ/edit#heading=h.v9bx55hrki7h).

Reply:
Comparing trying to import into an azure workspace before vs after my change: previously, it failed with the error "Workspace landing zone type [CromwellBaseResourcesFactory] does not support protected data" - but now seems like it hits my check first and gives "Unable to add protected snapshot to unprotected workspace".  If anyone here has knowledge or intuition about how desirable/undesirable this effect is, I'm willing to hear it - I can also check with product next week when everyone's back!